presto-main/src/test/java/io/prestosql/server/TestDynamicFilterService.java

```diff
@@ -339,17 +357,66 @@ public void testShortCircuitOnAllTupleDomain()
         assertTrue(dynamicFilter.isBlocked().isDone());
     }
 
+    @Test
+    public void testDynamicFilterCoercion()
+    {
+        DynamicFilterService dynamicFilterService = new DynamicFilterService(metadata, typeOperators, newDirectExecutorService());
+        DynamicFilterId filterId1 = new DynamicFilterId("df1");
+        SymbolAllocator symbolAllocator = new SymbolAllocator();
+        Symbol symbol1 = symbolAllocator.newSymbol("DF_SYMBOL1", INTEGER);
+        Expression df1 = new Cast(symbol1.toSymbolReference(), toSqlType(BIGINT));
+
+        QueryId queryId = new QueryId("query");
+        StageId stageId1 = new StageId(queryId, 1);
+
+        dynamicFilterService.registerQuery(
+                queryId,
+                session,
+                ImmutableSet.of(filterId1),
+                ImmutableSet.of(filterId1),
+                ImmutableSet.of());
+        dynamicFilterService.stageCannotScheduleMoreTasks(stageId1, 2);
```

Comment:
DF from single task is sufficient (test will be simpler). In this test we are testing coercions explicitly.